### DISCUSSION TITLE
Can I use custom plugins?

### DISCUSSION THREAD

**Discussion by apple-blossom** [qltysh/qlty#1179] [Q&A] [answered]
### Question

We are using custom linters and would like to use it with codeclimate. But I don't see new version including possibility to add custom plugins like is was [hehe](https://codeclimate.com/blog/build-your-own-codeclimate-plugin). How can I use qlty with a custom plugin? 

### CLI Version

0.446.0

- OS: 
Windows

**Comment by marschattha** [comment, marked as answer]
Actually the new version is more flexible and accommodating to adding custom linters.

You can check this [plugin guide](https://github.com/qltysh/qlty/blob/main/plugins/plugin_guide.md) as a reference although it is not comprehensive it should give you some idea. 

Here are examples of configuration already implemented linters: https://github.com/qltysh/qlty/tree/main/plugins/linters

You can use a custom source of plugin configuration: https://docs.qlty.sh/qlty-toml#sources or even define the entire plugin config in your qlty.toml file like in these [tests](https://github.com/qltysh/qlty/blob/main/qlty-cli/tests/cmd/check/driver_script_config_file_interpolation.in/.qlty/qlty.toml).

One limitation you might feel, depending on the output of your linter is parsing the output. We do support SARIF as standard and you can even write a [custom regex](https://github.com/qltysh/qlty/blob/main/qlty-cli/tests/cmd/check/regex_category_level.in/.qlty/qlty.toml), for more complex use cases I would recommend either converting your output to SARIF or a regex parsable output otherwise you will have to write a custom parser in Rust [like this](https://github.com/qltysh/qlty/blob/main/qlty-check/src/parser/rubocop.rs)
